### PR TITLE
Minor Changes 

### PR DESCRIPTION
**Pull request by sstoyanov7**
Improve readibility, assert messages and type hinting.

### CODE REVIEW

**Comment by hkdobrev** [comment]
@sstoyanov7 I've cherry-picked one commit, reworded it and released as 3.0.1 as it was a bug relased in 3.0.0: https://github.com/clippings/silex-pagination/commit/0682560a75c617895283f513e9509636950b7caa

On the other changes - I'd like to not diverge so much from the original as I would still like to see the original maintainer merge our changes so more people could make of them and use the service provider with Silex 2.x. I've pinged him earlier this week on his repo.

File: source/PaginationTrait.php
Comment (EZarkov) [inline source/PaginationTrait.php:30]: This is third type hint what this method returns, is it necessary?

Comment (sstoyanov7) [reply source/PaginationTrait.php:30]: Every call that returns unknown type is highly recommended to be specified.
This helps static analysis and finding bugs. Otherwise PHP my auto cast it to something that author didn't expected. Not to mention code understanding, code completion and debugging.

Particularly this fixed Travis CI issue. It's needed because that way PHPStan can compare the type after assignment and the function returned type.

**Comment by sstoyanov7** [comment]
@hkdobrev, thanks for this cherry pick below.
>@sstoyanov7 I've cherry-picked one commit, reworded it and released as 3.0.1 as it was a bug relased in 3.0.0: 0682560

>On the other changes - I'd like to not diverge so much from the original as I would still like to see the original maintainer merge our changes so more people could make of them and use the service provider with Silex 2.x. I've pinged him earlier this week on his repo.

The question is do these changes help development and developers, regardless if developers are original maintainers or maintainers who work on the fork?

My humble opinion is that it they help development and everyone. Not to mention that original has not been updated for a long time.

At the end is your call.

File: source/PaginationTrait.php
Comment (phgeorgiev) [reply source/PaginationTrait.php:30]: Is it necessary to have a variable and `@var` annotation since the method has a method return type specified?

**Review by phgeorgiev — CHANGES_REQUESTED** [review]

File: source/Tests/PaginationServiceProviderTest.php
Comment (phgeorgiev) [inline source/Tests/PaginationServiceProviderTest.php:26]: 2 spaces here.

File: source/Tests/PaginationServiceProviderTest.php
Comment (phgeorgiev) [inline source/Tests/PaginationServiceProviderTest.php:48]: Same here.

**Review by phgeorgiev — CHANGES_REQUESTED** [review]

**Comment by hkdobrev** [comment]
We're in the process of going away from Kilte pagination and using exclusively a single pagination package from KnpLabs. So I don't see us maintaining this fork.

**Pull request closed without merging**
